src/main/java/org/mule/extension/ftp/internal/command/FtpCommand.java

```diff
@@ -320,23 +320,44 @@ private Optional<FTPFile> doGetFileFromAbsoluteUri(URI absoluteUri) throws IOExc
   private Optional<FTPFile> getFileFromParentDirectory(URI absoluteUri) throws IOException {
     String filePath = normalizePath(absoluteUri.getPath());
     String fileParentPath = getParentPath(absoluteUri);
-    if (fileParentPath != null) {
-      if (tryChangeWorkingDirectory(fileParentPath)) {
-        FTPListParseEngine engine = client.initiateListParsing();
-        while (engine.hasNext()) {
-          FTPFile[] files = engine.getNext(FTP_LIST_PAGE_SIZE);
-          for (FTPFile file : files) {
-            if (file != null && FilenameUtils.getName(filePath).equals(file.getName())) {
-              return Optional.ofNullable(file);
-            }
-          }
+
+    if (fileParentPath == null) {
+      return Optional.of(createRootFile());
```

Comment:
We are risking a NullPointerException since the previous version had a `Optional.ofNullable`

Reply:
The [createRootFile](https://github.com/mulesoft/mule-ftp-connector/blob/2e561c1d7b5c1e3c534ea61a265d642667c12372/src/main/java/org/mule/extension/ftp/internal/command/FtpCommand.java#L384) method never returns null